code/game/objects/items/devices/flashlight.dm

```diff
@@ -98,6 +98,29 @@
 	brightness_on = 2
 	w_class = 1
 
+/obj/item/device/flashlight/heavy
+	name = "heavy duty flashlight"
+	desc = "A high-luminosity flashlight for specialist duties."
+	icon_state = "heavyflashlight"
+	item_state = "heavyflashlight"
+	brightness_on = 7

```

Comment:
Can you try lowering the value of owner_light_mult so that these aren't too powerful for diona? It affects the portion of the light that is cast on the holder's tile.

The default value is 0.5, try setting it as low as you can without it looking silly. You can test it ingame by setting that on the flashlight then turning it off and on again